validator/src/main/java/io/avaje/validation/core/adapters/NumberAdapters.java

```diff
@@ -220,12 +210,32 @@ public boolean isValid(Object value) {
       }
 
       final int sign = NumberSignHelper.signum(value, GREATER_THAN);
-      if (inclusive ? sign > 0 : sign >= 0) {
 
-        return false;
+      return !(inclusive ? sign > 0 : sign >= 0);
+    }
+  }
+
+  private static final class RangeAdapter implements ValidationAdapter<Object> {
+
+    private final ValidationAdapter<Number> adapter;
+
+    RangeAdapter(
+        ValidationContext.Message message, Set<Class<?>> groups, Map<String, Object> attributes) {
+
+      final var min = (int) attributes.get("min");
+      final var max = (int) attributes.get("max");
+      this.adapter =
+          new MaxAdapter(message, groups, max).andThen(new MinAdapter(message, groups, min));
```

Comment:
This produces difference error messages vs a range validator that produces a single range error.

Reply:
fixed